### PR TITLE
[#9459] Hamburger menu does not collapse

### LINKED ISSUE #9459: Hamburger menu does not collapse
- **Environment**:
`master` branch at commit 05be9b6c9. 

**Steps to reproduce**
1. Go to TEAMMATES home page in a small or medium size window 
2. Click on hamburger menu 
3. Click any of the options in the menu

**Expected behaviour**
The hamburger menu collapses after clicking a hyperlink.

**Actual behaviour**
The hamburger menu stays open and obstructs the content.

**Additional information**
Applicable to student, instructor and admin pages as well

### PR DESCRIPTION
**Pull request by amrut-prabhu**
Fixes #9459

**Outline of Solution**

Changes the navbar to collapsed when one of the list item anchors is clicked, or if the click occurs outside the expanded navbar

### CODE REVIEW

**Comment by amrut-prabhu** [comment]
@tanhengyeow Would you mind reviewing this?

**Comment by amrut-prabhu** [comment]
> I've not looked too deeply into this yet but may I know why a dedicated directive needs to be created to handle this behavior?

Not too familiar with Angular 😅 . This was the best Angular specific implementation that I could find online (apart from [angular-off-click](https://github.com/TheSharpieOne/angular-off-click)).

> Are there any good reasons why we can't use something similar to ngbDropdownToggle for the Hamburger menu in this case?

I'm trying to do it now, but facing some problems. Came across [dropdown sub menu items- issue](https://github.com/ng-bootstrap/ng-bootstrap/issues/768) (and more specifically, [this](https://github.com/twbs/bootstrap/issues/16387#issuecomment-97153831)) while doing so.

File: src/web/app/page.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<nav class="navbar navbar-expand-lg navbar-dark fixed-top">
+<nav class="navbar navbar-expand-lg navbar-dark fixed-top" (tmClickElsewhere)="isCollapsed = true">
```
Comment (tanhengyeow) [inline src/web/app/page.component.html]: `tmClickElsewhere` to `tmClickOutside` ?

File: src/web/app/page.component.html [hunk 2]
```diff
@@ -127,4 +127,4 @@ <h1 *ngIf="pageTitle && isValidUser">{{ pageTitle }}</h1>
       </div>
     </div>
   </footer>
-</div>
+</div>
```
Comment (tanhengyeow) [inline src/web/app/page.component.html]: Remember to leave an empty line :)

**Pull request merged**